### PR TITLE
Set Title to Current Panel

### PR DESCRIPTION
**Pull request by timmo001**
## Proposed change

Sets title to the current panel/section of the app. Uses the same mechanism the sidebar uses to get the title.

Screenshot_20200315_122647
Screenshot_20200315_123018
Screenshot_20200315_123039
Screenshot_20200315_123058
Screenshot_20200315_123133
Screenshot_20200315_123203
Screenshot_20200315_123230

## Type of change

- [x] New feature (thank you!)

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.

### CODE REVIEW

File: src/components/ha-sidebar.ts [hunk 1]
```diff
@@ -335,6 +335,27 @@ class HaSidebar extends LitElement {
         // @ts-ignore
         selectedEl.scrollIntoViewIfNeeded();
       }
+      this.setTitle();
+    }
+  }
+
+  private setTitle() {
```
Comment (balloob) [inline src/components/ha-sidebar.ts]: This is not something that should be managed by the sidebar. This should be a new state mixin for hass-element.

We should also extract the getPanelTitle function and move it to `data/panel.ts`

Comment (timmo001) [reply src/components/ha-sidebar.ts]: :ok_hand:

File: src/data/panel.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { HomeAssistant, PanelInfo } from "../types";
+import { DEFAULT_PANEL } from "../common/const";
+
+export const getPanelTitle = (hass: HomeAssistant): string | undefined => {
+  let title: string = "";
+  const panel = Object.values(hass.panels).find(
+    (p: PanelInfo): boolean => p.url_path === hass.panelUrl
+  );
+  if (panel) {
```
Comment (balloob) [inline src/data/panel.ts]: Can you make this a guard clause instead.

```ts
if (!panel) { return ""; }
```

**Review by balloob — APPROVED** [review]
Looks good. Ok to merge when last comment addressed! 👍

File: src/data/panel.ts (hunk 2 above)
Comment (timmo001) [reply src/data/panel.ts]: :ok_hand:

**Pull request merged**